pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -642,6 +642,44 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 			cdLog.Debug("not creating new provision since the deployment is set to try install only once")
 			return reconcile.Result{}, nil
 		}
+
+		if cd.Spec.LimitInstallRestarts != nil && cd.Status.InstallRestarts > *cd.Spec.LimitInstallRestarts {
+			cdLog.Debug("not creating new provision since the install restarts limit has reached")
+			conditions, changed := controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+				cd.Status.Conditions,
+				hivev1.ProvisionStoppedCondition,
+				corev1.ConditionTrue,
+				"InstallRestartsLimitReached",
+				fmt.Sprintf("Install restarts limit reached. Current limit is %v.", cd.Spec.LimitInstallRestarts),
```

Comment:
This message will include the address of the `LimitInstallRestarts` pointer and not the maximum number of restarts. Also, I don't know that it is necessary to include the limit in the condition message, given that the limit is directly available in the cd spec.